daemon/main.c

```diff
@@ -532,6 +532,28 @@ main_terminate(
 //  Set process scheduling policy/priority and Lock resident memory.
 //
 
+static int move_to_root_cgroup(void) {
+  FILE *f;
+
+  f = fopen("/sys/fs/cgroup/cpu/tasks", "w");
+  if (f == NULL) {
+    log_message(LOG_DEBUG, "Can't open cgroups tasks file for writing\n");
+    return(-1);
+  }
+
+  if (fprintf(f, "%d\n", getpid()) <= 0) {
```

Comment:
Only negative values are an error (see `RETURN VALUE` section of the manpage:https://man7.org/linux/man-pages/man3/fprintf.3p.html#RETURN_VALUE), but you are right that getting a value of `0` should be impossible as well: `fprintf` returns the number of characters printed on success, which can never be zero, so this can stay as it is.